File: TP1/arquivo.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n;
    printf("Digite o numero de valores reais a serem lidos: ");
    scanf("%d", &n);

    FILE *file = fopen("valores.txt", "w");

    // Lê e salva os valores sequencialmente no arquivo
    double valor;
    for (int i = 0; i < n; i++) {
        printf("Digite o valor real #%d: ", i + 1);
        scanf("%lf", &valor);
        fprintf(file, "%lf\n", valor);
    }

    // Fecha o arquivo
    fclose(file);

    // Reabre o arquivo para leitura reversa
    file = fopen("valores.txt", "r");

    // Move o ponteiro para o final do arquivo
    fseek(file, 0, SEEK_END);

    // Lê e mostra os valores de trás para frente
    printf("Valores lidos de tras para frente:\n");
    while (ftell(file) > 0) {
        fseek(file, -sizeof(double), SEEK_CUR);
        fread(&valor, sizeof(double), 1, file);
        printf("%lf\n", valor);
    }

    // Fecha o arquivo novamente
    fclose(file);

    return 0;
}

File: TP1/arquivo.java
import java.io.*;

public class arquivo {
    public static void main(String[] args) {
        try {
            
            System.out.print("Digite o número de valores reais a serem lidos: ");
            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            int n = Integer.parseInt(reader.readLine());

            
            RandomAccessFile file = new RandomAccessFile("valores.txt", "rw");

            
            for (int i = 0; i < n; i++) {
                System.out.print("Digite o valor real #" + (i + 1) + ": ");
                double valor = Double.parseDouble(reader.readLine());
                file.writeDouble(valor);
            }

           
            file.close();

            
            file = new RandomAccessFile("valores.txt", "r");

          
            file.seek(file.length());

            
            System.out.println("Valores lidos de trás para frente:");
            while (file.getFilePointer() > 0) {
                file.seek(file.getFilePointer() - 8);  
                double valor = file.readDouble();
                System.out.println(valor);
            }

            // Fecha o arquivo novamente
            file.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
